From f7f1622a9dc76d575b52c7f0463549be8444f357 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=98yvind=20Harboe?= <oyvind.harboe@zylin.com>
Date: Sat, 11 Apr 2026 16:32:32 +0200
Subject: [PATCH] 0013-tools-add-rtlil-stats-command-fix-stage-isolation-an

---
 flow/designs/asap7/gcd/config.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/flow/designs/asap7/gcd/config.mk b/flow/designs/asap7/gcd/config.mk
index c98498e7a..7c784df86 100644
--- a/flow/designs/asap7/gcd/config.mk
+++ b/flow/designs/asap7/gcd/config.mk
@@ -15,4 +15,3 @@ export PLACE_DENSITY          = 0.35
 # a smoketest for this option, there are a
 # few last gasp iterations
 export SKIP_LAST_GASP ?= 1
-
-- 
2.51.0

