From beb191192454f27022686b6993dd8640a56cb827 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=98yvind=20Harboe?= <oyvind.harboe@zylin.com>
Date: Sat, 11 Apr 2026 16:32:32 +0200
Subject: [PATCH] 0025-build-add-do-yosys-sdc-copy-target-for-parallel-synt

---
 flow/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/flow/Makefile b/flow/Makefile
index 5c8d5b16e..8b4526e00 100644
--- a/flow/Makefile
+++ b/flow/Makefile
@@ -276,6 +276,10 @@ do-yosys-canonicalize: yosys-dependencies
 do-yosys-keep: yosys-dependencies
 	$(SCRIPTS_DIR)/synth.sh $(SCRIPTS_DIR)/synth_keep.tcl $(LOG_DIR)/1_1_yosys_keep.log
 
+.PHONY: do-yosys-sdc-copy
+do-yosys-sdc-copy:
+	cp $(SDC_FILE) $(RESULTS_DIR)/1_2_yosys.sdc
+
 .PHONY: do-yosys-kept-json
 do-yosys-kept-json:
 	$(PYTHON_EXE) $(SCRIPTS_DIR)/rtlil_kept_modules.py $(RESULTS_DIR)/1_1_yosys_keep.rtlil $(RESULTS_DIR)/kept_modules.json
-- 
2.51.0

