From 20b078813d0d6b08e08a6ca82bea01d8c027291f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=98yvind=20Harboe?= <oyvind.harboe@zylin.com>
Date: Sat, 11 Apr 2026 16:32:33 +0200
Subject: [PATCH] 0029-fix-remove-mock-openroad-bazel_dep-after-inlining

---
 MODULE.bazel | 1 -
 1 file changed, 1 deletion(-)

diff --git a/MODULE.bazel b/MODULE.bazel
index 8c0217df4..811c222e5 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -6,7 +6,6 @@ module(
 )
 
 bazel_dep(name = "bazel-orfs")
-bazel_dep(name = "mock-openroad", version = "0.0.1")
 bazel_dep(name = "rules_python", version = "1.8.5")
 bazel_dep(name = "rules_shell", version = "0.6.1")
 
-- 
2.51.0

