diff --git a/flow/scripts/synth_preamble.tcl b/flow/scripts/synth_preamble.tcl
index 3edee2d..afd7d7e 100644
--- a/flow/scripts/synth_preamble.tcl
+++ b/flow/scripts/synth_preamble.tcl
@@ -44,7 +44,11 @@ proc read_design_sources { } {
   }
 
   if { [env_var_equals SYNTH_HDL_FRONTEND slang] } {
-    plugin -i slang
+    if { [info exists ::env(SLANG_PLUGIN_PATH)] } {
+      plugin -i $::env(SLANG_PLUGIN_PATH)
+    } else {
+      plugin -i slang
+    }
 
     set slang_args [list \
       -D SYNTHESIS --keep-hierarchy --compat=vcs --ignore-assertions --top $::env(DESIGN_NAME) \
