diff --git a/BUILD b/BUILD
new file mode 100644
index 0000000..0d57ccb
--- /dev/null
+++ BUILD
@@ -0,0 +1,3 @@
+package(
+	default_visibility = ["//visibility:public"],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..8419e9b
--- /dev/null
+++ MODULE.bazel
@@ -0,0 +1 @@
+module(name="external_pdk")
diff --git a/asap7/BUILD b/asap7/BUILD
new file mode 100644
index 0000000..c92202a
--- /dev/null
+++ asap7/BUILD
@@ -0,0 +1,10 @@
+package(
+	default_visibility = ["//visibility:public"],
+)
+filegroup(
+	name = "asap7",
+	srcs = glob([
+		"**"
+	]),
+    visibility = ["//visibility:public"],
+)
